lib/schema.js

```diff
@@ -157,18 +231,82 @@ const processView = (view, ms) => {
       let schema = ms.categories.get(view.Category);
       if (!schema) {
         schema =  ms.views.get(view.Category);
-        processView(schema, ms);
+        errors.push(...processView(view.Category, schema, ms));
       }
 
-      view.definition[key] = extractDefinition(schema.definition, prop, ms);
+      try {
+        view.definition[key] = extractByPath(
+          schema.definition,
+          prop,
+          ms,
+          name,
+          `Fields.${key}`
+        );
+      } catch (error) {
+        errors.push(error);
+      }
     }
   }
+
+  return errors;
 };
 
 const processViews = ms => {
-  for (const schema of ms.views.values()) {
-    processView(schema, ms);
+  const errors = [];
+  for (const [name, schema] of ms.views) {
+    errors.push(...processView(name, schema, ms));
   }
+  return errors.length === 0 ? null : new MetaschemaError(errors);
+};
+
+const processFields = (ms, category, fields, source, prop) => {
+  const errors = [];
+  for (const key in fields) {
+    if (!fields.hasOwnProperty(key)) {
+      continue;
+    }
+
+    const field = fields[key];
+
+    if (field.domain) {
+      const def = ms.domains.get(field.domain);
+      if (!def) {
+        errors.push(new SchemaValidationError(
+          'unresolvedDomain',
+          source,
+          `${prop}.${key}`,
+          { domain: field.domain }
+        ));
+      } else {
+        Object.assign(field, { domain: field.domain, definition: def });
+      }
+      continue;
```

Comment:
Nit: I think `else` would look better here.

Edit: Perhaps the second argument to the `Object.assign` can be extracted too and the call to it will be moved to the end of the function.